hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/config/StreamFileHealthIndicator.java

```diff
@@ -0,0 +1,154 @@
+package com.hedera.mirror.importer.config;
+
+/*-
+ * ‌
+ * Hedera Mirror Node
+ * ​
+ * Copyright (C) 2019 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.search.Search;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.boot.actuate.health.Status;
+
+import com.hedera.mirror.importer.domain.StreamType;
+import com.hedera.mirror.importer.parser.AbstractParserProperties;
+import com.hedera.mirror.importer.parser.balance.BalanceParserProperties;
+import com.hedera.mirror.importer.parser.event.EventParserProperties;
+import com.hedera.mirror.importer.parser.record.RecordParserProperties;
+
+public class StreamFileHealthIndicator implements HealthIndicator {
+    private static final String COUNT_KEY = "count";
+    private static final String LAST_CHECK_KEY = "lastCheck";
+    private static final String REASON_KEY = "reason";
+
+    private final boolean enabled;
+    private final Instant endDate;
+    private final MeterRegistry meterRegistry;
+    private final Duration streamFileStatusCheckBuffer;
+    private final StreamType streamType;
```

Comment:
Same principle applies here. Please don't cache variables that are encapsulated in another class. There's no performance penalty for using the property getter. Also we may want to support these changing during runtime.
```suggestion
    private final MeterRegistry meterRegistry;
    private final ParserProperties parserProperties;
```

Reply:
Updated